test/python/unittest/API/APITest.py

```diff
@@ -0,0 +1,326 @@
+
+    def testDatatype(self):
+        """ Test openPMD.Datatype. """
+        data_type = openPMD.Datatype(1)
+
+    def testDataset(self):
+        """ Test openPMD.Dataset. """
+        data_type = openPMD.Datatype(1)
+        extent = openPMD.Extent()
+        obj = openPMD.Dataset(data_type, extent)
+
+    def testExtent(self):
+        """ Test openPMD.Extent. """
+        obj = openPMD.Extent()
+
+    def testGeometry(self):
+        """ Test openPMD.Geometry. """
+        obj = openPMD.Geometry(0)
+
+    def testIteration(self):
+        """ Test openPMD.Iteration. """
+        self.assertRaises(TypeError, openPMD.Iteration)
+
+        iteration = self.__particle_series.iterations[400]
+
+        copy_iteration = openPMD.Iteration(iteration)
+
+        self.assertIsInstance(copy_iteration, openPMD.Iteration)
+
+    def testIteration_Container(self):
+        """ Test openPMD.Iteration_Container. """
+        obj = openPMD.Iteration_Container()
+
+    def testIteration_Encoding(self):
+        """ Test openPMD.Iteration_Encoding. """
+        obj = openPMD.Iteration_Encoding(1)
+
+    def testMesh(self):
+        """ Test openPMD.Mesh. """
+        self.assertRaises(TypeError, openPMD.Mesh)
+        mesh = self.__series.iterations[100].meshes['E']
+        copy_mesh = openPMD.Mesh(mesh)
+
+        self.assertIsInstance(copy_mesh, openPMD.Mesh)
+
+    def testMesh_Container(self):
+        """ Test openPMD.Mesh_Container. """
+        obj = openPMD.Mesh_Container()
+
+    def testParticlePatches(self):
+        """ Test openPMD.ParticlePatches. """
+        self.assertRaises(TypeError, openPMD.ParticlePatches)
+
+    def testParticleSpecies(self):
+        """ Test openPMD.ParticleSpecies. """
+        self.assertRaises(TypeError, openPMD.ParticleSpecies)
+
+    def testParticle_Container(self):
+        """ Test openPMD.Particle_Container. """
+        obj = openPMD.Particle_Container()
+
+    def testRecord(self):
+        """ Test openPMD.Record. """
+        # Has only copy constructor.
+        self.assertRaises(TypeError, openPMD.Record)
+
+        ### FIXME
+        ## Get a record (fails)
+        #record = self.__series.iterations[100].particles['electrons'].properties['positions'].components['x']
+
+        # Copy.
+        #copy_record = openPMD.Record(record)
+
+        # Check.
+        #self.assertIsInstance(copy_record, openPMD.Record)
+
+    def testRecord_Component(self):
+        """ Test openPMD.Record_Component. """
+        self.assertRaises( TypeError, openPMD.Record_Component)
+
+
+    def testFieldRecord(self):
+        """ Test querying for a non-scalar field record. """
+
+        E = self.__series.iterations[100].meshes["E"]
+        Ex = E["x"]
+
+        print (type(Ex))
+        self.assertIsInstance(Ex, openPMD.Mesh_Record_Component)
+
+
+
+# TODO: add __getitem__ to openPMD.Mesh and openPMD.Particle object for
+#       non-scalar records: return a record component
+# E_x = i.meshes["E"]["x"]
```

Comment:
you can do that now, see the example here :-)
https://github.com/openPMD/openPMD-api/blob/0.2.0-alpha/examples/2_read_serial.py#L34-L44